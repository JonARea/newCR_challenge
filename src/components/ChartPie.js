import React from 'react'
import {PieChart, Pie, Cell, Tooltip, Legend} from 'recharts'
import {setChartSize} from '../utils'
import PropTypes from 'prop-types'

const ChartPie = ({data, colors, category}) => {
  const dataArray = data ? Object.keys(data).map((catName, index) => ({
    name: catName,
    value: data[catName],
    color: colors[index % colors.length]
  })) : null

  return (dataArray ?
    <div>
      <h2>{`Percentages by ${category}`}</h2>
      <PieChart
        width={window.innerWidth}
        height={window.innerHeight - 150}
        >
        <Legend />
        <Pie
          data={dataArray}
          dataKey="value"
          nameKey="name"
          cx={'50%'}
          cy={'50%'}
          outerRadius={setChartSize()}
          fill="#8884d8"
          label
          >
          { dataArray &&
            dataArray.map(elem => (
              <Cell fill={elem.color} key={elem.name}>
                {elem.name}
              </Cell>))
          }
        </Pie>
        <Tooltip />
      </PieChart>
    </div> : <h2>Please upload some data</h2>
  )
}

ChartPie.propTypes = {
  data: PropTypes.object,
  colors: PropTypes.array.isRequired,
  category: PropTypes.string.isRequired
}

export default ChartPie
