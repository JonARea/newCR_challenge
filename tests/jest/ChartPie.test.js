import React from 'react'
import renderer from 'react-test-renderer'
import ChartPie from '../../src/components/ChartPie'
import {ProcessedUserData} from '../userData'
const data = ProcessedUserData.percentageByGender
const colors = ['#333', '#eee']
const category = 'gender'

test('Pie Charts render correctly', () => {
  const component = renderer.create(
    <ChartPie
      data={data}
      colors={colors}
      category={category}
    />)

  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()

})
