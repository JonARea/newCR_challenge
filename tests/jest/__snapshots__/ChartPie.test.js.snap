// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pie Charts render correctly 1`] = `
<div>
  <h2>
    Percentages by gender
  </h2>
  <div
    className="recharts-wrapper"
    style={
      Object {
        "cursor": "default",
        "height": 618,
        "position": "relative",
        "width": 1024,
      }
    }
  >
    <svg
      className="recharts-surface"
      height={618}
      style={undefined}
      version="1.1"
      viewBox="0 0 1024 618"
      width={1024}
    >
      <clipPath
        id="recharts1-clip"
      >
        <rect
          height={608}
          width={1014}
          x={5}
          y={5}
        />
      </clipPath>
      <g
        className="recharts-layer recharts-pie"
      >
        <g
          clipPath="url(#recharts-pie-2)"
        >
          <g
            className="recharts-layer"
          >
            <g
              className="recharts-layer recharts-pie-sector"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            />
            <g
              className="recharts-layer recharts-pie-sector"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            />
          </g>
        </g>
      </g>
    </svg>
    <div
      className="recharts-legend-wrapper"
      style={
        Object {
          "bottom": 5,
          "height": "auto",
          "left": 5,
          "position": "absolute",
          "width": 1014,
        }
      }
    >
      <ul
        className="recharts-default-legend"
        style={
          Object {
            "margin": 0,
            "padding": 0,
            "textAlign": "center",
          }
        }
      >
        <li
          className="recharts-legend-item legend-item-0"
          style={
            Object {
              "display": "inline-block",
              "marginRight": 10,
            }
          }
        >
          <svg
            className="recharts-surface"
            height={14}
            style={
              Object {
                "display": "inline-block",
                "marginRight": 4,
                "verticalAlign": "middle",
              }
            }
            version="1.1"
            viewBox="0 0 32 32"
            width={14}
          >
            <path
              className="recharts-legend-icon"
              d="M0,4h32v24h-32z"
              fill="#333"
              stroke="none"
            />
          </svg>
          <span
            className="recharts-legend-item-text"
          >
            females
          </span>
        </li>
        <li
          className="recharts-legend-item legend-item-1"
          style={
            Object {
              "display": "inline-block",
              "marginRight": 10,
            }
          }
        >
          <svg
            className="recharts-surface"
            height={14}
            style={
              Object {
                "display": "inline-block",
                "marginRight": 4,
                "verticalAlign": "middle",
              }
            }
            version="1.1"
            viewBox="0 0 32 32"
            width={14}
          >
            <path
              className="recharts-legend-icon"
              d="M0,4h32v24h-32z"
              fill="#eee"
              stroke="none"
            />
          </svg>
          <span
            className="recharts-legend-item-text"
          >
            males
          </span>
        </li>
      </ul>
    </div>
    <div
      className="recharts-tooltip-wrapper"
      style={
        Object {
          "MozTransform": "translate(undefinedpx, undefinedpx)",
          "OTransform": "translate(undefinedpx, undefinedpx)",
          "WebkitTransform": "translate(undefinedpx, undefinedpx)",
          "msTransform": "translate(undefinedpx, undefinedpx)",
          "pointerEvents": "none",
          "position": "absolute",
          "top": 0,
          "transform": "translate(undefinedpx, undefinedpx)",
          "visibility": "hidden",
        }
      }
    >
      <div
        className="recharts-default-tooltip"
        style={
          Object {
            "backgroundColor": "#fff",
            "border": "1px solid #ccc",
            "margin": 0,
            "padding": 10,
            "whiteSpace": "nowrap",
          }
        }
      >
        <p
          className="recharts-tooltip-label"
          style={
            Object {
              "margin": 0,
            }
          }
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;
